### PR TITLE
Update references to shared objects as supported, link to related mat…

### PR DESCRIPTION
**Pull request by Clay-Mysten**
…erials

### CODE REVIEW

File: doc/src/build/move/sui-move-library.md [hunk 1]
```diff
@@ -70,16 +70,14 @@ transfer::freeze_object(obj);
...
-### Share an object (experimental)
-This feature is still in development. It only works in Move for demo purpose, and doesn't yet work in Sui.
-
+### Share an object
 To make an object `obj` shared and mutable, one can call:
 ```
 transfer::share_object(obj);
 ```
-After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be deleted, transferred or embedded in another object as a field.
+After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be deleted, transferred or embedded in another object as a field. For an example, see [Shared Object](https://examples.sui.io/basics/shared-object.html#shared-object) on https://examples.sui.io/.
```
Comment (sblackshear) [inline doc/src/build/move/sui-move-library.md:78]:
```suggestion
After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be transferred or embedded in another object as a field. For more detail, see the [shared objects](../../learn/build/objects.md#shared) documentation.
```

- I moved the more thorough description of shared objects to the "objects" section and simply linked to that description here
- We will soon (but do not yet) support shared object deletion, so I think we can avoid talking about deletion for now + explicitly mention that is supported once it lands.

File: doc/src/build/objects.md [hunk 2]
```diff
@@ -33,11 +33,13 @@ struct A {
...
+
+Here is a brief primer: for any other mutable object that's not shared, no two transactions can be mutating the same object at the same time since a transaction is pinned on a specific (and must be the latest) version of each object. Hence, Sui doesn't need to worry about reaching a consensus or ordering (it's ordered by construction). However for shared objects, in order to allow multiple transactions to mutate the same object at the same time, we need a sequencer or [consensus engine](../../learn/architecture/consensus.md) to properly order these transactions. Because of this, using a shared object can be much more expensive in terms of latency, throughput and gas cost. On the other hand, a shared object is also a powerful primitive that allows for expressing richer behavior that doesn't require central trust.
```
Comment (sblackshear) [inline doc/src/build/objects.md:42]:
```suggestion
```

File: doc/src/build/objects.md [hunk 3]
```diff
@@ -33,11 +33,13 @@ struct A {
...
+### Shared
+An object can be shared, meaning that anyone can use and mutate this object. For an example, see [Shared Object](https://examples.sui.io/basics/shared-object.html#shared-object) on https://examples.sui.io/. To see how use of shared objects affects your applications, see the two implementations of TicTacToe: [No Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/tic_tac_toe.move) vs. [Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/shared_tic_tac_toe.move).
```
Comment (sblackshear) [inline doc/src/build/objects.md:40]:
```suggestion
An object can be shared, meaning that anyone can read or write this object. In contrast to mutable owned objects (which are single-writer), shared objects require [consensus](../../learn/architecture/consensus.md) to sequence reads and writes. For an example of creating and accessing a shared object, see [Shared Object](https://examples.sui.io/basics/shared-object.html#shared-object) on https://examples.sui.io/.

In other blockchains, every object is shared. However, Sui programmers often have the choice to implement a particular use-case using shared objects, owned objects, or a combination. This choice can have implications for performance, security, and implementation complexity The best way to understand these tradeoffs is to look at a few examples of use-cases implemented both ways:

Escrow: [Shared](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/defi/sources/shared_escrow.move), [Owned](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/defi/sources/escrow.move)
Auction: [Shared](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/nfts/sources/shared_auction.move), [Owned](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/nfts/sources/auction.move)
Tic Tac Toe: [Shared](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/shared_tic_tac_toe.move), [Owned](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/tic_tac_toe.move)
```

File: doc/src/build/move/sui-move-library.md [hunk 4]
```diff
@@ -70,16 +70,14 @@ transfer::freeze_object(obj);
...
+Shared mutable objects can be powerful in that they make programming a lot simpler in many cases. However, shared objects are also more expensive to use: they require a full sequencer (a.k.a. a [consensus engine](../../learn/architecture/consensus.md) to order the transactions that touch the shared object, which means longer latency/lower throughput and higher gas cost. One can see the difference in the two programming schemes by looking at the two implementations of TicTacToe: [No Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/tic_tac_toe.move) vs. [Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/shared_tic_tac_toe.move).
```
Comment (sblackshear) [inline doc/src/build/move/sui-move-library.md:80]:
```suggestion
```

**Review by sblackshear — COMMENTED** [review]
Thank you Clay!

File: doc/src/build/move/sui-move-library.md (hunk 1 above)
Comment (Clay-Mysten) [reply doc/src/build/move/sui-move-library.md:78]: Thanks, Sam!

**Review by randall-Mysten — APPROVED** [review]

**Pull request merged**